tests/test_docstrings.py

```diff
@@ -0,0 +1,98 @@
+import inspect
```

Comment:
See https://github.com/samuelcolvin/watchfiles/blob/main/tests/test_docs.py for (IMHO) a better implementation.

But don't worry about this too much as we'll probably need to take that logic and move it into a separate pypi package so we can use it in both this repo and pydantic itself.

Reply:
Ah `@pytest.mark.parametrize('module_name,source_code', generate_code_chunks('watchfiles', 'docs'))` is very neat! I'll leave the test file as is for now, and I can maybe change to this when the rest of the PR is ready.

Reply:
are these tests currently running the examples? If so leave them like this for now.

Reply:
yes, they're currently running the examples